openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -180,7 +181,8 @@ def create_account_with_params(request, params):
     if skip_email:
         registration.activate()
     else:
-        compose_and_send_activation_email(user, profile, registration)
+        data = get_params_for_activation_email(request, registration, user)
```

Comment:
Let's please change this method `get_params_for_activation_email` to prepare data and send the email as well, In this way we can reduce the core change.